BE/app/domain/user/user_service.py

```diff
@@ -9,8 +9,8 @@ class UserService:
     """User service defines a repository interface for user entity."""
     user_repo = UserRepository()
 
-    def create(self, regis: RegisterRequest) -> bool:
-        user_dto = self.user_repo.create(regis)
+    def create(self, regis: RegisterRequest, db) -> bool:
```

Comment:
`db` should be a dependency of Repository rather than this function.
Try to use https://github.com/google/pinject
to maintain IoC

Reply:
Hi a Paul,
So I will create a db repository class, then define an add method ? For example:  
`
class DbRepository:`
`
 def add(object):`

Reply:
Currently, I follow this tutorial about dependencies inject:
https://fastapi.tiangolo.com/tutorial/dependencies/
https://fastapi.tiangolo.com/tutorial/sql-databases/
My flow is:
`def login(self, login_req: LoginRequest,
              db: Session = Depends(deps.get_db)):`
        `user = self.auth_service.login(login_req.email, login_req.password, db)`
then
`def login(self, email: str, password: str,
              db
              ) -> Optional[Auth]:`
       ` # Returns jwt token`
        `auth_user = self.auth_repo.query_auth_user(email, db)`
and finally, db is used by `auth_repo`